### PR TITLE
[gardening] Clean repeated use of force unwrapping on test files

### PR DESCRIPTION
**Pull request by otaviolima**

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

File: TestFoundation/TestISO8601DateFormatter.swift [hunk 1]
```diff
@@ -31,74 +31,74 @@ class TestISO8601DateFormatter: XCTestCase {
     func test_stringFromDate() {
         let formatter = DateFormatter()
         formatter.dateFormat = "yyyy/MM/dd HH:mm zzz"
-        let someDateTime = formatter.date(from: "2016/10/08 22:31 GMT")
+        let someDateTime = formatter.date(from: "2016/10/08 22:31 GMT")!
         let isoFormatter = ISO8601DateFormatter()
```
[the remark this answers is not in the document]
Comment (otaviolima) [reply TestFoundation/TestISO8601DateFormatter.swift]: That is true, I will do it.

**Comment by otaviolima** [comment]
I don't get why my GPG status is `unverified` gonna check that and try to fix, if this is a problem.

**Comment by spevans** [comment]
@swift-ci please test

File: TestFoundation/TestISO8601DateFormatter.swift [hunk 2]
```diff
@@ -179,118 +182,128 @@ class TestISO8601DateFormatter: XCTestCase {
...
-        XCTAssertEqual(ISO8601DateFormatter.string(from: someDateTime!, timeZone: timeZone!, formatOptions: formatOptions), "10-W40-06")
+        XCTAssertEqual(ISO8601DateFormatter.string(from: someDateTime, timeZone: timeZone, formatOptions: formatOptions), "10-W40-06")
 
 #if !os(Android)
         /*
          The following tests cover various cases when changing the .formatOptions property with a different TimeZone set.
          */
 
-        timeZone = TimeZone(identifier: "PST")
+        guard let pstTimeZone = TimeZone(identifier: "PST") else {
```
[the remark this answers is not in the document]
Comment (spevans) [reply TestFoundation/TestISO8601DateFormatter.swift]: I don't think one failing test should abort all of the testing, even if in this case something fundamental like "PST"  is missing. At least with `XCTFail()` better failure messaging can be provided (which should be fixed here). We don't consider any failing `XCTAssert*()` should lead to a `fatalError`.

**Comment by otaviolima** [comment]
Rebased and added the messages to `XCTFail()` calls.

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**